torch_xla/csrc/tensor.cpp

```diff
@@ -136,35 +136,59 @@ std::vector<xla::util::Cleanup> LockDevices(const std::set<Device>& devices) {
   return unlocker;
 }
 
-struct TensorHasher {
-  size_t operator()(const at::Tensor& tensor) const {
-    return xla::util::HashCombine(xla::util::GetEnumValue(tensor.scalar_type()),
-                                  TensorHash(tensor));
+class XlaDataCacheArena {
+ public:
+  struct TensorHasher {
+    size_t operator()(const at::Tensor& tensor) const {
+      return xla::util::HashCombine(
+          xla::util::GetEnumValue(tensor.scalar_type()), TensorHash(tensor));
+    };
   };
-};
-struct TensorComparer {
-  bool operator()(const at::Tensor& tensor1, const at::Tensor& tensor2) const {
-    return tensor1.scalar_type() == tensor2.scalar_type() &&
-           tensor1.equal(tensor2);
+  struct TensorComparer {
+    bool operator()(const at::Tensor& tensor1,
+                    const at::Tensor& tensor2) const {
+      return tensor1.scalar_type() == tensor2.scalar_type() &&
+             tensor1.equal(tensor2);
+    }
+  };
+
+  using XlaDataCache =
+      xla::util::Cache<at::Tensor, xla::ComputationClient::Data, TensorHasher,
+                       TensorComparer>;
+
+  explicit XlaDataCacheArena(size_t max_cache_size)
+      : max_cache_size_(max_cache_size) {}
+
+  XlaDataCache* Get(const Device& device) {
+    std::lock_guard<std::mutex> lock(mutex_);
+    auto it = device_caches_.find(device);
+    if (it == device_caches_.end()) {
+      it = device_caches_.emplace(device, new XlaDataCache(max_cache_size_))
```

Comment:
You need to emplace an unique pointer here, this is why the build fails.

Reply:
Done.
Weird, was working for me on my workstation.

Reply:
CircleCI uses a different compiler.